extensions/commands/deploy-lipo/cmd_deploy_lipo.py

```diff
@@ -0,0 +1,60 @@
+import os
+import shutil
+from subprocess import run
+
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conan.cli.args import common_graph_args
+from conan.errors import ConanException
+
+from _lipo import lipo as lipo_folder
+
+
+_valid_archs = [
+    'x86',
+    'x86_64',
+    'armv7',
+    'armv8',
+    'armv8_32',
+    'armv8.3',
+    'armv7s',
+    'armv7k'
+]
+
+@conan_command(group="Consumer")
+def deploy_lipo(conan_api: ConanAPI, parser, *args):
+    """
+    Deploy dependencies for multiple profiles and lipo into universal binaries
+    """
+    common_graph_args(parser)
+    parsed = parser.parse_args(*args)
+    profiles = parsed.profile_host or parsed.profile
+    if not profiles:
+        raise ConanException("Please provide profiles with -pr or -pr:h")
+    other_args = []
+    i = 0
+    while i < len(args[0]):
+        arg = args[0][i]
+        if arg in ['-pr', '-pr:h', '--profile', '--profile:host']:
+            i += 2
+        else:
+            other_args.append(arg)
+            i += 1
+    for profile in profiles:
+        run(['conan', 'install',
+             '--deploy=full_deploy',
+             '-pr:h', profile
+             ] + other_args)
+    output_dir = os.path.join('full_deploy', 'host')
+    for package in os.listdir(output_dir):
```

Comment:
Maybe the README needs a better explanation of the purpose.  The goal is to produce Conan built libraries that can be turned into universal libraries and used outside of Conan.  This should be as simple as possible, without trying to make universal packages or modifying recipes or toolchains.  The '--deploy=full_deploy' option seems to be the same idea for single architecture situations.

The other complex point that was missed in the discussions under some other threads: this isn't just about supporting two or more architectures, but really multiple profiles.  For example, I'd say it's pretty common in Xcode on macOS to target os.version=11.0 for armv8 (first supported OS) and something like os.version=10.13 for x86_64.

Since Conan packages are by design built around a single arch, os.version, etc. how do we do this in a single conan process?  The sequence of commands should look like:

```
conan install --deploy=full_deploy -pr:h x86_64 . # Creates full_deploy/host/zlib/1.2.13/Release/x86_64/lib/libz.a
conan install --deploy=full_deploy -pr:h armv8 . # Creates full_deploy/host/zlib/1.2.13/Release/armv8/lib/libz.a
lipo full_deploy/host/zlib/1.2.13/Release/x86_64/lib/libz.a full_deploy/host/zlib/1.2.13/Release/armv8/lib/libz.a -create -output full_deploy/host/zlib/1.2.13/Release/lib/libz.a
# copy include files, etc to universal path as well
```
Now you can create an Xcode project outside of Conan with multiple architectures and references to files under 'full_deploy'.  The conanfile.py is really only necessary to declare the dependencies, and maybe have build() run xcodebuild on the pre-existing Xcode project.

Any thoughts about how that fits as a custom command, or does this need to be a wrapper script outside of Conan?

Reply:
The above is good when ``conan install --requires=zlib/1.2.13``, because it has no dependencies. But I am thinking about the case for example of ``conan install --requires=boost/1.80``, where boost depends on zlib and bzip2. So the ``full_deploy`` will deploy the 3 libraries, boost, zlib, and bzip2. What would be the expectaction/desired behavior in this case? To create a fat library for the 3 of them? like one fat boost, another fat zlib and another bzip2?

Reply:
Yes, exactly.  In the example project under 'xcode' I'm depending on "libtiff/4.5.0" and the deployment produces:
jbig
libdeflate
libjpeg
libtiff
libwebp
xz_utils
zlib
zstd

This is why posting an example somewhere with an Xcode project is going to be really helpful.

Reply:
I was thinking about a different approach that involves a deployer instead of the custom command weirdness (accepting all install command line arguments, and calling conan recursively).  It would involve using lipo to replace a single arch at a time.  The user would then run:
```
conan install -d lipo -pr x86_64 -b missing . # Creates 'full_deploy' as single arch universal binaries
conan install -d lipo -pr armv8 -b missing . # Runs 'lipo -replace' on 'full_deploy' to add the armv8 arch
conan build . # or xcodebuild
```
Does this seem like a better approach?  (Let me make sure it's possible before closing this PR). It would be less efficient with a large number of archs but even on iOS it's just a few.